src/coreclr/jit/codegen.h

```diff
@@ -1530,13 +1530,19 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         }
     };
 
-    OperandDesc genOperandDesc(GenTree* op);
+    OperandDesc genOperandDesc(GenTree* op, insOpts instOptions = INS_OPTS_NONE, var_types simdBaseType = TYP_UNKNOWN);
```

Comment:
Do we still need to track the `simdBaseType` or is the fact that the broadcast/constant is `contained` "enough" (with the constant having been marked as `fromScalar`)?

I'd imagine we would have validated in lowering that the size was correct and not done the containment/marking with flag otherwise.

Reply:
`simdBaseType` is used to emit the constant scalar when the operand is a constant vector, since we don't have the base type information without `simdBaseType`.

Reply:
If we did https://github.com/dotnet/runtime/pull/84821#discussion_r1185371882, we could handle this via the contained broadcast instead. That would simplify things and avoid using one of the few flags we have remaining. It would also cover the other broadcast scenarios that are relevant for small constants.